Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardSettingsViewController.swift

```diff
@@ -171,20 +156,33 @@ class CreditCardSettingsViewController: UIViewController, Themeable {
 
     // MARK: - Private helpers
 
-    private func updateStateForEditView() {
-        guard appAuthenticator.canAuthenticateDeviceOwner() else { return }
+    private func updateStateForEditView(editState: CreditCardEditState) {
+        // Update credit card edit view state before showing
+        viewModel.addEditViewModel.state = editState
+        creditCardEditView = CreditCardEditView(
+            viewModel: viewModel.addEditViewModel,
+            dismiss: { [weak self] successVal in
+                DispatchQueue.main.async {
+                    if successVal {
+                        self?.updateCreditCardList()
+                    }
+                    self?.creditCardAddEditView?.dismiss(animated: true)
+                    self?.viewModel.addEditViewModel.clearValues()
+                }
+        })
 
-        appAuthenticator.authenticateWithDeviceOwnerAuthentication { result in
-            switch result {
-            case .success:
-                self.creditCardAddEditView.view.isHidden = false
-            case .failure:
-                self.logger.log("Failed to authenticate", level: .debug, category: .creditcard)
-            }
-        }
+        guard let creditCardEditView = creditCardEditView else { return }
+        creditCardAddEditView = UIHostingController(rootView: creditCardEditView)
+        creditCardAddEditView?.view.backgroundColor = .clear
+        creditCardAddEditView?.modalPresentationStyle = .formSheet
+        present(creditCardAddEditView!, animated: true, completion: nil)
     }
 
     @objc private func addCreditCard() {
         updateState(type: .add)
     }
+
+    private func editCreditCard() {
+        updateState(type: .view)
```

Comment:
This is a placeholder as we actually need to pass on the full credit card details from `CreditCardSettingsViewController` -> `CreditCardEditView` for fully editing of the credit card. 

Another way to do it is to pass the guid